code/modules/research/xenobiology/crossbreeding/warping

```diff
@@ -0,0 +1,796 @@
+	RegisterSignal(T,COMSIG_ATOM_EXITED,.proc/angry_out)
+
+
+/obj/effect/warped_rune/redspace/proc/angry_in()
+
+	if(rage) //don't buff anyone if someone is already buffed by the rune.
+		return
+
+	for(var/mob/living/carbon/human/A in T)
+		og_punch_low = A.dna.species.punchdamagelow
+		og_punch_high = A.dna.species.punchdamagehigh
+		og_threshold = A.dna.species.punchstunthreshold
+		H = A
+		RegisterSignal(H,COMSIG_HUMAN_MELEE_UNARMED_ATTACK,.proc/blind_rage)//triggers whenever H attacks
+		H.dna.species.punchdamagelow = 40
+		H.dna.species.punchdamagehigh = 50
+		H.dna.species.punchstunthreshold = 50 //so each punch doesn't insta stun
+		rage = TRUE
+
+
+/obj/effect/warped_rune/redspace/proc/angry_out()
+	var/turf/HT = get_turf(H)
+	if(HT == T && locate(runepath) in T)
+		return
+	H.dna.species.punchdamagelow = og_punch_low
+	H.dna.species.punchdamagehigh = og_punch_high
+	H.dna.species.punchstunthreshold = og_threshold 
+	UnregisterSignal(H,COMSIG_HUMAN_MELEE_UNARMED_ATTACK)
+	rage = FALSE //allows another person to enter rage mode on the rune.
+
+///triggers whenever the enraged player punches something. lowers their punch damage and deals stamina damage to themselves.
+/obj/effect/warped_rune/redspace/proc/blind_rage()
+	H.dna.species.punchdamagelow -= 10
+	H.dna.species.punchdamagehigh -= 10
+	H.dna.species.punchstunthreshold -= 10
+	H.adjustStaminaLoss(10)
+	to_chat(H, "<span class='warning'>You are feeling weaker with each punch!</span>")
+//this proc has a bug where it triggers even when you pick up an item with an empty hand.
+
+
+///destroying the rune will also remove the punch force of the person currently in rage mode
+/obj/effect/warped_rune/redspace/Destroy()
+	..()
+	angry_out()
+
+
+
+
+
+
+
+
+/* Green rune vores plasma and spews out xeno resin which lets you build xeno structure*/
+
+/obj/item/slimecross/warping/green
+	colour = "green"
+	runepath = /obj/effect/warped_rune/greenspace
+
+/obj/effect/warped_rune/greenspace
+	icon_state = "xeno_rune"
+	desc = "We will build walls out of our fallen foes, they shall fear our very buildings."
+	var/obj/item/stack/sheet/xeno_resin/X
+	max_cooldown = 100
+
+
+/obj/item/stack/sheet/xeno_resin
+	name = "Resin sheets"
+	icon = 'icons/mob/alien.dmi'
+	icon_state = "nestoverlay" //literally just the xeno nest icon
+	merge_type = /obj/item/stack/sheet/xeno_resin
+
+
+/obj/effect/warped_rune/greenspace/Initialize()
+	.=..()
+	START_PROCESSING(SSprocessing, src)
+
+/obj/effect/warped_rune/greenspace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		transmute_resin()
+
+/obj/effect/warped_rune/greenspace/proc/transmute_resin()
+	for(var/obj/item/stack/sheet/mineral/plasma/P in T)
+		X = new(T)
+		P.amount--
+		if(P.amount <= 0)
+			qdel(P)
+
+//list of things you can build with resin sheets.
+GLOBAL_LIST_INIT(resin_recipes, list ( \
+	new/datum/stack_recipe("Resin seed", /obj/structure/alien/weeds/node, 10, one_per_turf = 1, on_floor = 1, time = 100), \
```

Comment:
inconsistent capitalization in the names of these recipes

Reply:
also, maybe you could consider allowing people to craft resin into certain xeno organs and/or xeno hides? like, don't let people craft egg sacs or neurotoxin spitters, but the other xeno organs (resin spinners, xeno tongues, hive nodes, xeno eyes, xeno livers, etc.) should be fine. if you think that it doesn't make sense that resin should be allowed to be made into that stuff, consider maybe adding a new xeno material for if you put leather, bones, sinew, hides, or other semi-rare organic material on this rune.

Reply:
also, you should add resin windows and resin nests to this list, since those are also structures that xenos can make